src/stcal/dq_init/dq_initialization.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import numpy as np
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+def do_dqinit(input_model, mask_model, guider_list=None):
+    """Perform the dq_init step on a datamodel
+
+    Parameters
+    ----------
+    input_model : input science datamodel
+        The science datamodel to be corrected
+
+    mask_model : mask datamodel
+        The mask model to use in the correction
+
+    Returns
+    -------
+    output_model : science datamodel
+        The dq corrected science datamodel
+    """
+
+    # Inflate empty DQ array, if necessary
+    check_dimensions(input_model)
+
+    # Create output model as copy of input
+    output_model = input_model.copy()
```

Comment:
I wonder if this copy is really necessary but this is a discussion for the JWST group.

Reply:
That's been debated back and forth over the years. The purpose of the copy is so that when people run the step interactively from within python their input data does *not* get overwritten/updated in place by the output, but instead returns a separate output. It does lead to extra memory use, unfortunately. At one point we contemplated having a step-level param that would indicate whether the input should be updated in place or a new output created.

Reply:
The update-in-place approach could be the default for use in Ops, where no one cares about the intermediate results of individual steps that are passed along within a pipeline module.